--- /Users/user165258/Downloads/xulsword/Cpp/CMakeLists.txt	2020-03-14 17:29:24.000000000 -0700
+++ /Users/user165258/Downloads/xulsword.orig/Cpp/CMakeLists.txt	2020-03-13 17:38:25.000000000 -0700
@@ -6,7 +6,7 @@
 set(xulswordProj_VERSION_REVISION "4")
 
 # -Wl,--no-undefined flags are required by GCC to make it check shared library dependencies
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wl,--no-undefined -pthread")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wl,-undefined,error -pthread")
 
 # sword does not install all the sword headers, so use the local include dir
 include_directories(./src/include ./sword/include  ${CMAKE_INCLUDE_PATH})
