diff -ru sword.64win/CMakeLists.txt sword-new/CMakeLists.txt
--- sword.64win/CMakeLists.txt	2022-11-30 17:40:44.698731851 -0800
+++ sword-new/CMakeLists.txt	2022-11-30 17:53:44.546698415 -0800
@@ -316,3 +316,14 @@
 IF(SWORD_BUILD_TESTS STREQUAL "Yes")
 	ADD_SUBDIRECTORY("${CMAKE_CURRENT_SOURCE_DIR}/tests")
 ENDIF(SWORD_BUILD_TESTS STREQUAL "Yes")
+
+# For Linux cross-compilation to Windows
+IF(CMAKE_TOOLCHAIN_FILE MATCHES "cmake-toolchain.64win")
+  add_definitions(-DWIN32 -D_WIN32 -DBUILDING_LIBRARY)
+  add_library(libwsock32 SHARED IMPORTED)
+  find_library(SOCKLIB libwsock32 NAMES libwsock32.a PATHS "/usr/x86_64-w64-mingw32/lib")
+  message(STATUS "${SOCKLIB}")
+  target_link_libraries(sword ${SOCKLIB})
+  target_link_libraries(sword_static ${SOCKLIB})
+  set(CMAKE_CXX_FLAGS "-fpermissive ${CMAKE_CXX_FLAGS}")
+ENDIF(CMAKE_TOOLCHAIN_FILE MATCHES "cmake-toolchain.64win")
diff -ru sword.64win/include/ftplib.h sword-new/include/ftplib.h
--- sword.64win/include/ftplib.h	2022-11-30 17:40:49.410731649 -0800
+++ sword-new/include/ftplib.h	2022-11-30 17:52:19.574702058 -0800
@@ -94,7 +94,6 @@
 #endif /* (_FTPLIB_NO_COMPAT) */
 /* end v1 compatibility stuff */
 
-GLOBALREF int ftplib_debug;
 GLOBALREF void FtpInit(void);
 GLOBALREF char *FtpLastResponse(netbuf *nControl);
 GLOBALREF int FtpConnect(const char *host, netbuf **nControl);
