diff -ru CMakeLists.txt CMakeLists.txt
--- CMakeLists.txt	2025-06-02 14:21:05.000000000 -0700
+++ CMakeLists.txt	2025-06-17 14:17:30.654227846 -0700
@@ -276,13 +276,14 @@
 ##############################################################################################
 # Our build test
 #
-
-ADD_EXECUTABLE(buildtest buildtest.cpp)
-IF(BUILDING_STATIC)
-	TARGET_LINK_LIBRARIES(buildtest sword_static)
-ELSE(BUILDING_STATIC)
-	TARGET_LINK_LIBRARIES(buildtest sword)
-ENDIF(BUILDING_STATIC)
+IF(SWORD_BUILD_TESTS)
+  ADD_EXECUTABLE(buildtest buildtest.cpp)
+  IF(BUILDING_STATIC)
+    TARGET_LINK_LIBRARIES(buildtest sword_static)
+  ELSE(BUILDING_STATIC)
+    TARGET_LINK_LIBRARIES(buildtest sword)
+  ENDIF(BUILDING_STATIC)
+ENDIF(SWORD_BUILD_TESTS)
 
 ##############################################################################################
 # Installing the library, headers, utilies, etc
@@ -323,3 +324,25 @@
 	INCLUDE(CTest)
 	ADD_SUBDIRECTORY("${CMAKE_CURRENT_SOURCE_DIR}/tests")
 ENDIF(SWORD_BUILD_TESTS)
+
+# For Linux cross-compilation to Windows
+IF(CMAKE_TOOLCHAIN_FILE MATCHES "windows")
+  # Use same CLucene compilation flags:
+  set(CMAKE_CXX_FLAGS "-fvisibility=hidden -fvisibility-inlines-hidden -O2 -g -DNDEBUG -D_REENTRANT -D_UCS2 -D_UNICODE ${CMAKE_CXX_FLAGS}")
+  add_definitions(-DWIN32 -D_WIN32 -DBUILDING_LIBRARY)
+  
+  # Windows requires that the type of import target library (dll or a) be
+  # decided by headers. CLucene shared headers require the definition of
+  # MAKE_CLUCENE_XX_LIB to allow import from an .a type library.
+  target_compile_definitions(sword_static PUBLIC -DMAKE_CLUCENE_CORE_LIB -DMAKE_CLUCENE_SHARED_LIB)
+  
+  add_library(libwsock32 SHARED IMPORTED)
+  find_library(SOCKLIB libwsock32 NAMES libwsock32.a PATHS "/usr/x86_64-w64-mingw32/lib")
+  message(STATUS "${SOCKLIB}")
+  IF(TARGET sword_static)
+    target_link_libraries(sword_static ${SOCKLIB})
+  ENDIF(TARGET sword_static)
+  IF(TARGET sword)
+    target_link_libraries(sword ${SOCKLIB})
+  ENDIF(TARGET sword)
+ENDIF(CMAKE_TOOLCHAIN_FILE MATCHES "windows")
